stopcovid/clients/sqs.py

```diff
@@ -1,20 +1,46 @@
 import boto3
 import os
 import json
-import uuid
+from typing import List
+
+from marshmallow import Schema, fields, post_load
 
 SQS = boto3.resource("sqs")
 
 
-def publish_outbound_sms_messages(payloads):
+class OutboundSMSSchema(Schema):
+    event_id = fields.Str(required=True)
+    phone_number = fields.Str(required=True)
+    body = fields.Str(required=True)
+
+    @post_load
+    def make_outbound_sms(self, data, **kwargs):
+        return OutboundSMS(**data)
+
+
+class OutboundSMS:
+    def __init__(self, event_id: str, phone_number: str, body: str):
+        self.event_id = event_id
+        self.phone_number = phone_number
+        self.body = body
+
+
+def publish_outbound_sms_messages(outbound_sms_messages: List[OutboundSMS]):
     queue_name = f"outbound-sms-{os.getenv('STAGE')}"
     queue = SQS.get_queue_by_name(QueueName=queue_name)
 
-    entries = []
-    for payload in payloads:
-        # TODO: use event_id once we have a callsite for this method
-        idempotency_key = str(uuid.uuid4())
-        payload["idempotency_key"] = idempotency_key
-        entries.append({"Id": idempotency_key, "MessageBody": json.dumps(payload)})
+    entries = [
+        {
+            "Id": outbound_sms.event_id,
+            "MessageBody": json.dumps(
+                {
+                    "To": outbound_sms.phone_number,
+                    "Body": outbound_sms.body,
+                    "idempotency_key": outbound_sms.event_id,
```

Comment:
Consider using message attributes to store the idempotency key instead of embedding it in the message body. https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-attributes.html